### ISSUE TITLE
Multi Page Form 400 Bad Request - No submission exists with ID

### ISSUE THREAD

**Issue by jwrudzin** [verbb/formie#562]
**Description**
I have a form with 4 pages. When I submit page 3, I get a message "Couldn’t save submission due to errors. 400: Bad Request".

When using AJAX submission, in the network tab of dev tools I can see the first and second pages submitting properly. Then the third page responds with a 400 error (No submission exists with the ID):

![400 response](https://user-images.githubusercontent.com/18213990/131859942-7181dac5-dd4d-4bf7-b4b0-d931f5f3ed05.jpg)

One thing to note is that after I submit the first page of the form, I can see my submission under "incomplete" in the CMS. However, after I submit the second page of the form the incomplete submission disappears.

One other thing to note is that I'm using conditionals for the third and fourth pages of the form, but disabling those conditionals did not fix this issue. I wanted to note this in case setting them up initially is what led to this issue.

I also tried using server-side submission instead of AJAX, but still ran into an issue at the same part of the form, although in this case I got back the error message text I have set on the form instead of the "bad request" error.

One other thing to note is that this appears to be a bit inconsistent - I was able to get a couple of submissions through initially, but now I'm able to reproduce this issue repeatedly.

**Steps to reproduce**
1. Create a form with 4 pages.
2. Add fields to each page.
3. Set conditionals for showing the third and fourth pages.
4. Attempt to submit the form.

### Form settings
- Multi-page form: Yes
- Submission Method: Ajax
- Client-side Validation: Yes
- Custom Form Templates: No

**Additional info**
- Plugin version: 1.4.14
- Craft version: 3.7.11
- Multi-site: No

**Comment by jwrudzin** [comment]
After some additional testing here I was able to track down some additional information on this issue. I noticed that the success of the submission seemed to be tied to the content I was entering into the fields. 

I figured out that the problem was that one of the words I had typed into a text field on the second page was matching one of the spam keywords I had entered in settings. Once I removed that spam keyword, I was able to get the submission to go through using the same content.

This is of course not desired behavior when a submission is flagged as spam halfway through a multi-page form, because it doesn't allow the user to finish completing the form, keeps no record of the submission in the CMS (the submissions were not moved to spam from incomplete, I double checked), and is very confusing for the user who is trying to submit the form and getting an irrelevant error message.

Just to be clear, some updated steps for reproducing this issue are:

1. Create a form with multiple pages.
2. Add fields to each page.
3. Add some spam keywords in settings.
4. Attempt to submit an early page of the form with a spam keyword entered into one of the fields.

**Comment by engram-design** [comment]
Yep so that error message is when you try to submit a form, using a submission that's no longer incomplete. It'll try to find an incomplete, currently active submission to add content to (for each subsequent page). Seeing as the submission is moved from incomplete to complete after the second page shows that this is happening.

It almost certainly sounds like a conditional page issue, where it thinks submitting on page 2 is the last page of the form. But as you note taking those out don't seem to make any difference.

Yes, this is a great point about flagging as spam for an incomplete submission. Spam checks and captchas are actually enabled upon each step of a multi-page form. There have been some people who specifically want spam checks to have happen on each step of a form. Mostly for use-cases where a single page is quite long, or is an application process where users fill out on page at a time, at different sessions.

I believe we'll keep the flagging as spam as-is (so it's flagged as soon as a spammy value is encountered, no matter the page), but you're correct that it shouldn't kick out a user as soon as a submission is marked as spam, if it's not on the final submission (where it'll follow settings to mark as success, and move to spam). That's just horrible UX - didn't realise it was doing that behaviour!

Fixed for the next release. To get the fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.4.14",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [1.4.15](https://github.com/verbb/formie/releases/tag/1.4.15)
